pkg/auth/policy.go

```diff
@@ -0,0 +1,268 @@
+package auth
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"golang.org/x/crypto/openpgp"
+
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/util"
+)
+
+// A Policy encapsulates the behavior a p2 node needs to authorize
+// its actions. It is possible for implementations to rely on other
+// services for these behaviors, so these calls may be slow or
+// transiently fail.
+type Policy interface {
+	// Check if a pod is authorized to be installed and run on this
+	// node. This involves checking that the pod has a valid signature
+	// and that the signer is authorized to install/run the pod. If
+	// the action is authorized, `nil` will be returned.
+	AuthorizePod(
+		id string,
```

Comment:
The way `pod.Verify()` is implemented, a temp directory is created, files are saved to it, then the files are re-read to perform the verification.  Is there any reason to keep this design?  If not, I'd prefer to re-write how it gathers its signature data to expose an interface similar to what I did with the `pod.Manifest`: expose a `SignatureData()` method that returns the data needed to verify the signature.